src/test/java/seedu/address/logic/commands/DeleteCommandTest.java

```diff
@@ -76,6 +79,49 @@ public void execute_invalidIndexFilteredList_throwsCommandException() {
         assertCommandFailure(deleteCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
     }
 
+    @Test
+    public void execute_validFullName_success() {
+        Person personToDelete = TypicalPersons.BENSON;
+        DeleteCommand deleteCommand = new DeleteCommand(new Name("Benson Meier"));
+
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_PERSON_SUCCESS, personToDelete);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.deletePerson(personToDelete);
+
+        assertCommandSuccess(deleteCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_validNameMultipleResults_throwsCommandException() {
+        DeleteCommand deleteCommand = new DeleteCommand(new Name("Benson"));
+        model.addPerson(new PersonBuilder().withName("Benson Meiere")
+                .withAddress("311, Clementi Ave 2, #02-25")
+                .withEmail("johnd@example.com").withPhone("98765432")
+                .withTags("owesMoney", "friends").build());
+
+        assertCommandFailure(deleteCommand, model, Messages.MESSAGE_MULTIPLE_PERSON_DISPLAYED_NAME);
+    }
+
+    @Test
+    public void execute_validNameNoResults_throwsCommandException() {
+        DeleteCommand deleteCommand = new DeleteCommand(new Name("Ben"));
+
+        assertCommandFailure(deleteCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_NAME);
+    }
+
+    @Test
+    public void execute_validName_success() {
+        Person personToDelete = TypicalPersons.BENSON;
+        DeleteCommand deleteCommand = new DeleteCommand(new Name("Benson"));
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_PERSON_SUCCESS, personToDelete);
+
```

Comment:
Good job on handling the cases for the different full name inputs for the updated `delete` command